vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/model/implementation/Serializer.java

```diff
@@ -384,9 +386,13 @@ private static String addQuotes(final String string) {
 		return QUOTE + string + QUOTE;
 	}
 
-
 	private static String addAngleBrackets(final String string) {
 		return LESS_THAN + string + MORE_THAN;
 	}
 
+	public static String getFactOutput(Predicate predicate, List<Term> terms) {
+		return predicate.getName() + terms.toString().replace(OPENING_BRACKET, OPENING_PARENTHESIS)
```

Comment:
Also, when serializing a Fact, we do more changes (call `getIRIString` on the predicate name).
Please make sure the result of the chase can be round-tripped (i.e. read as a set of equivalent facts).
This may require more processing when the result includes blanks.